netmiko/base_connection.py

```diff
@@ -551,6 +552,19 @@ def is_alive(self) -> bool:
                 # If unable to send, we can tell for sure that the connection is unusable
                 return False
         return False
+    
+    def is_session_closed(self) -> bool:
+        """
+        Returns a boolean flag with the state of the connection
+        this is a cafy custom function needed because in cisco_vxr_Ssh code uses self.remote_conn.closed
+        attribute to check connection status. However the 'closed' attriubute is present on remote_conn object only if protocol is ssh
+        and not telnet. Therefore to handle telnet remote_Conn, this function is created a wrapper
+        """
+        if self.protocol == "telnet":
+            return not self.is_alive()
+        else:
+            return self.remote_conn.closed
+        
```

Comment:
In next PR ,the code should be 
```suggestion
    def is_session_closed(self) -> bool:
        """
        Returns a boolean flag with the state of the connection
        this is a cafy custom function needed because in cisco_vxr_Ssh code uses self.remote_conn.closed
        attribute to check connection status. However the 'closed' attriubute is present on remote_conn object only if protocol is ssh
        and not telnet. Therefore to handle telnet remote_Conn, this function is created a wrapper
        """
        if self.protocol == "telnet":
            return not self.sock._closed
        else:
            return self.remote_conn.closed
        
```